ci/docker/build.Dockerfile

```diff
@@ -15,7 +15,7 @@ RUN apt-get -yqq update && \
     apt-get -yqq install --no-install-recommends \
     software-properties-common \
     build-essential gfortran \
-    autoconf automake ninja-build pkg-config \
+    autoconf automake libssl-dev ninja-build pkg-config \
```

Comment:
This is required because with newer versions of spack `openssl` is detected as an external but the header files are not there without `libssl-dev`, leading to build failures later.